src/transformer.ts

```diff
@@ -205,9 +205,10 @@ function combineInlineSourceMaps(
 function combineSourceMaps(
     program: ts.Program, filePath: string, tscSourceMapText: string): string {
   const tscSourceMap = parseSourceMap(tscSourceMapText);
+  const fileDir = path.dirname(filePath);
   let tscSourceMapGenerator: SourceMapGenerator|undefined;
   for (const sourceFileName of tscSourceMap.sources) {
-    const sourceFile = program.getSourceFile(sourceFileName);
+    const sourceFile = program.getSourceFile(path.resolve(fileDir, sourceFileName));
```

Comment:
The original tsickle_compiler_host already does this, I just reimplemented it incorrectly.